### PR TITLE
Fix for Issue#51: Instance Reference Claims and Instance Endorsement Claims

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**

### CODE REVIEW

**Comment by yogeshbdeshpande** [comment]
This is just a sample change, to display the concept we discussed today.

This has both Instance Endorsed and Instance Reference Claims properly extensible!

Needs a tidy up further, which we be done on Tuesday!

**Comment by nedmsmith** [comment]
Thanks Yogesh! What is the reasoning for when the extension points would be used in reference-claim-map-extension vs. ref-claim-value-map-extension?

**Comment by yogeshbdeshpande** [comment]
> Thanks Yogesh! What is the reasoning for when the extension points would be used in reference-claim-map-extension vs. ref-claim-value-map-extension?

reference-claim-map-extension point would be exercised only when adding new keys-values to
reference-claim-map, for example a new parameter that also qualifies to be part of the reference claim something more
on top of reference values. In terms of CDDL context also it does not seem the
correct point to extend the existing reference values.

Once the proposed fix in Issue #53 is debated and if agreed, I will rebase my changes 
on top of that fix. Essentially the base concept on Fix for Issue #51 and Issue #53
are the same, only difference been issue#51 extends the same concept to 
instance reference and instance endorsement claims.

**Review by nedmsmith — COMMENTED** [review]
Once PR#56 is applied, this PR should be revisited.

File: comid-code-points.cddl [hunk 1]
```diff
@@ -6,6 +6,12 @@ comid.entity = 3
 comid.linked-tags = 4
 comid.claims = 5
 
+; ref-instance-claim-map
+comid.ref-claim-value = 1
+
+; endors-instance-claim-map
+comid.endors-claim-value = 1
```
Comment (thomas-fossati) [inline comid-code-points.cddl:13]: Both would clash with `comid.mac-addr`.  The first usable code-point is 6 (please check it).

File: concise-mid-tag.cddl [hunk 2]
```diff
@@ -191,6 +199,7 @@ endorsed-value-map = non-empty<{
   * $$endorsed-value-map-extension
 }>
 
+
```
Comment (thomas-fossati) [inline concise-mid-tag.cddl:202]: spurious newline?

**Review by thomas-fossati — COMMENTED** [review]
I suggest we make the naming a bit more consistent with the rest of the document.  Concretely:
* `end` -> `endorsed`
* `endors` -> `endorsed`
* `ref` -> `reference`

File: comid-code-points.cddl (hunk 1 above)
Comment (yogeshbdeshpande) [reply comid-code-points.cddl]: Have verified and new available code point is 6 so started from their.

**Comment by yogeshbdeshpande** [comment]
> I suggest we make the naming a bit more consistent with the rest of the document. Concretely:
> 
> * `end` -> `endorsed`
> * `endors` -> `endorsed`
> * `ref` -> `reference`

Well the names became a bit long, but I agree with consistency aspect hence corrected to align logically with
reference and endorsed and also endorsements on key names consistent with the existing names!

File: comid-code-points.cddl [hunk 3]
```diff
@@ -104,6 +98,12 @@ comid.serial-number = 3
 comid.ueid = 4
 comid.uuid = 5
 
+; reference-instance-claim-map
+comid.reference-claim-value = 6
+
+; endorsed-instance-claim-map
+comid.endorsed-claim-value = 7
```
Comment (thomas-fossati) [inline comid-code-points.cddl:105]: should be 6 as well since it's embedded in a different map

File: comid-code-points.cddl [hunk 4]
```diff
@@ -89,8 +83,8 @@ comid.element-value = 1
 comid.reference-claims = 0
 comid.endorsements = 1
 comid.identity-claims = 2
-comid.ref-instance-claims = 3
-comid.endors-instance-claims = 4
+comid.reference-instance-claims = 3
+comid.endorsements-instance-claims = 4
```
Comment (thomas-fossati) [inline comid-code-points.cddl:87]: maybe s/endorsements/endorsed/ ?  (and change L84 as well)

Comment (yogeshbdeshpande) [reply comid-code-points.cddl:87]: No as I said this is to maintain consistency, will retain like this for now to be consistent with the rest of the variables...

File: comid-code-points.cddl (hunk 3 above)
Comment (yogeshbdeshpande) [reply comid-code-points.cddl:105]: Good point they are on a different map, so can be same! I will correct it!

File: comid-code-points.cddl (hunk 4 above)
Comment (thomas-fossati) [reply comid-code-points.cddl:87]: np.  I've opened #58 to track this

Comment (yogeshbdeshpande) [reply comid-code-points.cddl:87]: Thanks, we can change the consistency aspect globally on Issue #58. For now we can proceed with this change!

**Review by thomas-fossati — APPROVED** [review]

**Comment by yogeshbdeshpande** [comment]
Ned, Henk, I assume you do not have any comments! I shall wait for an hour prior to commit? 
Any one of you please approve as well!

**Pull request merged**